.github/workflows/install.yaml

```diff
@@ -0,0 +1,51 @@
+name: Install
```

Comment:
I don't think that we need a dedicated action to run the install script. All e2e runs in CI will use the install script so I would expect install script issues to be present in e2e runs.

Reply:
Unfortunately, that doesn't seem accurate. We’ve been addressing issues with the script for exactly this reason. For example, the error in [PR #1429](https://github.com/operator-framework/operator-controller/pull/1429) would have prevented the CI from passing. **Could you please check and let me know where this test might be duplicated elsewhere?**

Additionally, we are validating the build/docker process locally as well. This validation currently cannot be tested in any other environment because it’s broken and fails. Error Examples: (before changes):  make docker-build failed on both macOS and Linux due to a hardcoded ./bin/linux path that does not exist. Here’s an example of the error: [Error example](https://github.com/operator-framework/operator-controller/actions/runs/11814261205/job/32912965441) Even if we create the dir it will still failing [see error](https://github.com/operator-framework/operator-controller/actions/runs/11814618324/job/32914032322)

Reply:
The Makefile as it is on 'main' right now works fine for me on Linux. The test you linked to above is failing because it appears it was before you changed the docker-build line to target the ./bin directory rather than ./bin/linux.

I don't have access to a mac but I don't really see how the changes in this PR are really different from how it worked before--it seems to just be changing where we build the binaries from ./bin/linux to ./bin? And it's also effectively removing the functionality of the existing 'go-build-local' target and replacing it with the current 'go-build-linux,' just renaming them and removing the former since you're still setting GOOS=linux.

It is interesting that the existing go-build-local on main isn't directly used anywhere--I'm assuming it's there for cases where somebody wanted to compile locally and would just locally edit the docker-build target to use 'build' for one-off testing. Maybe we should just make another docker-build target that uses -local, i.e. docker-build-local, if that's a workflow we see people needing/wanting

Reply:
I agree with @everettraven. We are (in theory) automatically generating a manifest and rendering an install script as part of the e2e.

If there are gaps in what happens with the installation during the e2e, let's fix it in the e2e instead of introducing a separate test.

Reply:
@camilamacedo86 Part of running our e2e tests includes doing, more or less, what you are doing here.

If you look at the `test-e2e` make target at https://github.com/operator-framework/operator-controller/blob/6d79092c8921f888b125e084b494576e87d5706e/Makefile#L179-L188 and dig even further into the `run` target at https://github.com/operator-framework/operator-controller/blob/6d79092c8921f888b125e084b494576e87d5706e/Makefile#L279-L280 you can see that it runs the make targets `docker-build`, `kind-cluster`, `kind-load`, and `kind-deploy`. The `kind-deploy` target at https://github.com/operator-framework/operator-controller/blob/6d79092c8921f888b125e084b494576e87d5706e/Makefile#L222-L226 uses the same installation script that we generate for releases.

Essentially, everything you are doing in this CI action is already done as part of all of our e2e runs in CI. 

I'm not quite sure why CI didn't catch the issues you linked, but I also never ended up running into those problems. Is it possible that there may be some difference in tooling versions that resulted in you experiencing those issues?

Reply:
HI @everettraven 

>  docker-build kind-cluster kind-load kind-deploy 

I see that is calling so that is true. 
Thank you for point out. 

>  Is it possible that there may be some difference in tooling versions that resulted in you experiencing those issues?

Not that is not really possible 
if you revert https://github.com/operator-framework/operator-controller/pull/1429 OR https://github.com/operator-framework/operator-controller/pull/1411 you will see that the error occurs always.

**So probably we move forward besides the e2e failures.**